internal/app/machined/internal/phase/config/extra_files.go

```diff
@@ -32,15 +35,59 @@ func (task *ExtraFiles) runtime(r runtime.Runtime) (err error) {
 	var result *multierror.Error
 
 	for _, f := range r.Config().Machine().Files() {
-		p := filepath.Join("/var", f.Path)
+		// Slurp existing file if append is our op and add contents to it
+		var existingFileContents []byte
+		if f.Op == "append" {
+			existingFileContents, err = slurpFile(f.Path)
+			if err != nil {
+				result = multierror.Append(result, err)
+				continue
+			}
+
+			f.Contents = string(existingFileContents) + "\n" + f.Contents
+		}
+
+		// Determine if supplied path is in /var or not.
+		// If not, we'll write it to /var anyways and bind mount below
+		p := f.Path
+		inVar := true
+		explodedPath := strings.Split(
+			strings.TrimLeft(f.Path, "/"),
+			string(os.PathSeparator),
+		)
+
+		if explodedPath[0] != "var" {
+			p = filepath.Join("/var", f.Path)
+			inVar = false
+		}
+
 		if err = os.MkdirAll(filepath.Dir(p), os.ModeDir); err != nil {
 			result = multierror.Append(result, err)
+			continue
 		}
 
 		if err = ioutil.WriteFile(p, []byte(f.Contents), f.Permissions); err != nil {
 			result = multierror.Append(result, err)
+			continue
+		}
+
+		// File path was not /var/... so we assume a bind mount is wanted
+		if !inVar {
+			if err = unix.Mount(p, f.Path, "", unix.MS_BIND|unix.MS_RDONLY, ""); err != nil {
+				result = multierror.Append(result, fmt.Errorf("failed to create bind mount for %s: %w", p, err))
+			}
 		}
 	}
 
 	return result.ErrorOrNil()
 }
+
+// slurpFile simplyu reads in a file and returns a byte slice
+func slurpFile(path string) ([]byte, error) {
+	file, err := ioutil.ReadFile(path)
```

Comment:
isn't this function just `ioutil.ReadFile()` - it does nothing else?

Reply:
yep - that's fair. I thought it was going to be more complex than it was :)

will pull that up into the other code